File: app/src/androidTest/java/com/elyeproj/javadaggeruitest/ApplicationTestModule.java
package com.elyeproj.javadaggeruitest;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import static org.mockito.Mockito.mock;

@Module
public class ApplicationTestModule {
    protected final Application mApplication;
    public ApplicationTestModule(Application application) {
        mApplication = application;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @Singleton
    InjectedData provideInjectedData() {
        return mock(InjectedData.class);
    }
}

File: app/src/androidTest/java/com/elyeproj/kotlinespressomock/ApplicationTestModule.kt
package com.elyeproj.kotlinespressomock

import android.app.Application

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import org.mockito.Mockito.mock

@Module
class ApplicationTestModule(protected val mApplication: Application) {

    @Provides
    fun provideApplication(): Application {
        return mApplication
    }

    @Provides
    @Singleton
    fun provideInjectedData(): InjectedData {
        return mock(InjectedData::class.java)
    }
}